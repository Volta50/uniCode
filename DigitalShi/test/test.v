module sumadorOneBit(
    input A,B,Cin,
    output Co,So);

    wire x_ab;
    wire cout_t;
    wire a_ab;



    xor(x_ab,A,B);
    and(a_ab,A,B);
    and(cout_t,x_ab,Cin);
    or(Co, cout_t,a_ab);
    xor(So,x_ab,Cin);

endmodule