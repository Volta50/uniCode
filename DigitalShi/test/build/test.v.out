#! /usr/bin/vvp
:ivl_version "12.0 (stable)" "(v12_0-dirty)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/usr/lib/ivl/system.vpi";
:vpi_module "/usr/lib/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib/ivl/v2005_math.vpi";
:vpi_module "/usr/lib/ivl/va_math.vpi";
:vpi_module "/usr/lib/ivl/v2009.vpi";
S_0x5616fc857570 .scope package, "$unit" "$unit" 2 1;
 .timescale 0 0;
S_0x5616fc857700 .scope module, "sumdaorOneBit" "sumdaorOneBit" 3 1;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "A";
    .port_info 1 /INPUT 1 "B";
    .port_info 2 /INPUT 1 "Cin";
    .port_info 3 /OUTPUT 1 "Co";
    .port_info 4 /OUTPUT 1 "So";
o0x7fc3f6572018 .functor BUFZ 1, C4<z>; HiZ drive
o0x7fc3f6572048 .functor BUFZ 1, C4<z>; HiZ drive
L_0x5616fc8a09b0 .functor XOR 1, o0x7fc3f6572018, o0x7fc3f6572048, C4<0>, C4<0>;
L_0x5616fc8a0aa0 .functor AND 1, o0x7fc3f6572018, o0x7fc3f6572048, C4<1>, C4<1>;
o0x7fc3f6572078 .functor BUFZ 1, C4<z>; HiZ drive
L_0x5616fc8a0be0 .functor AND 1, L_0x5616fc8a09b0, o0x7fc3f6572078, C4<1>, C4<1>;
L_0x5616fc8a0ca0 .functor XOR 1, L_0x5616fc8a09b0, o0x7fc3f6572078, C4<0>, C4<0>;
L_0x5616fc8a0d90 .functor OR 1, L_0x5616fc8a0be0, L_0x5616fc8a0aa0, C4<0>, C4<0>;
v0x5616fc857890_0 .net "A", 0 0, o0x7fc3f6572018;  0 drivers
v0x5616fc8a03c0_0 .net "B", 0 0, o0x7fc3f6572048;  0 drivers
v0x5616fc8a0480_0 .net "Cin", 0 0, o0x7fc3f6572078;  0 drivers
v0x5616fc8a0550_0 .net "Co", 0 0, L_0x5616fc8a0d90;  1 drivers
v0x5616fc8a0610_0 .net "So", 0 0, L_0x5616fc8a0ca0;  1 drivers
v0x5616fc8a06d0_0 .net "a_ab", 0 0, L_0x5616fc8a0aa0;  1 drivers
v0x5616fc8a0790_0 .net "cout_t", 0 0, L_0x5616fc8a0be0;  1 drivers
v0x5616fc8a0850_0 .net "x_ab", 0 0, L_0x5616fc8a09b0;  1 drivers
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "-";
    "test.v";
